whylog_vim/input_reader/__init__.py

```diff
@@ -1,17 +1,48 @@
 import re
-from whylog_vim.input_reader.consts import Input, RegexPatterns
 
+from whylog.constraints import IdenticalConstraint
+from whylog.constraints.const import ConstraintType
 
-def filter_comments(content):
-    return [line for line in content if not RegexPatterns.COMMENTS.match(line)]
+from whylog_vim.input_reader.consts import RegexPatterns, ConstraintInput
+from whylog_vim.utils import get_id_from_name
 
 
-def get_button_name(line, offset):
-    offset -= 1
-    for match in re.finditer(RegexPatterns.BUTTON, line):
-        if offset >= match.start(0) and offset < match.end(0):
-            return match.group(0)[1:-1]
+class InputReader():
+    @classmethod
+    def filter_comments(cls, content):
+        return [line for line in content if not RegexPatterns.COMMENTS.match(line)]
 
+    @classmethod
+    def get_button_name(cls, line, offset):
+        offset -= 1
+        for match in re.finditer(RegexPatterns.BUTTON, line):
+            if offset >= match.start(0) and offset < match.end(0):
+                return match.group(0)[1:-1]
 
-def prepare_regex(pattern):
-    return ('^' + pattern + '$') % Input.EMPTY_GROUP
+    @classmethod
+    def parse_primary_key_groups(cls, content):
+        return map(int, content[0].split(', '))
+
+
+class ConstraintReader():
+
+    CONSTRAINTS = {ConstraintType.IDENTICAL: IdenticalConstraint,}
+
+    @classmethod
+    def _parse_constraint(cls, lines):
+        constr_type = ConstraintInput.TYPE.match(lines[0]).group(1)
+        groups = []
+        for line in range(1, len(lines)):
+            match = ConstraintInput.GROUP.match(lines[line])
+            if match:
+                groups.append((get_id_from_name(match.group('content1')), int(match.group('int2'))))
```

Comment:
Maybe move 'content1' and 'int2' to consts? Give them proper name: what means `content1` and `int2`?